yann/utils/raster.py

```diff
@@ -98,7 +98,7 @@ def tile_raster_images(X, img_shape, tile_shape, tile_spacing=(0, 0),
                                     dtype='uint8')
         else:
             out_array = numpy.zeros((out_shape[0], out_shape[1], 4),
-                                    dtype=X.dtype)
+                                    dtype=X[1].dtype)
```

Comment:
Again, are you sure of this ? Check this too please on an actual visualizer code.